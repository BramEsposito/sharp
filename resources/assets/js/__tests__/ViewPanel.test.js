import { mount } from '@vue/test-utils';
import ViewPanel from '../components/list/ViewPanel.vue';

describe('view-panel', ()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = mount(ViewPanel);
    });

    test('mount ViewPanel', ()=>{
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount "visible" ViewPanel', ()=>{
        wrapper.setProps({ show: true });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount "with content" ViewPanel', ()=>{
        wrapper.setProps({ show: true, content: 'SOME HTML MARKUP' });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('emit "change" to false on clicked background glasspane', ()=>{
        wrapper.setProps({ show: true });
        let glasspane = wrapper.find('.SharpViewPanel__glasspane');
        glasspane.trigger('click');
        expect(wrapper.emitted().change).toEqual([[false]]);
    });

});