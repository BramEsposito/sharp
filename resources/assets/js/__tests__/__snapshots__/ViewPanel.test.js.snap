// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view-panel mount "visible with content" ViewPanel 1`] = `
<div>
	<div class="SharpViewPanel__glasspane" style=""></div>
	<div class="SharpViewPanel" style="" enter-class="SharpViewPanel--collapsed" enter-active-class="SharpViewPanel--expanding" enter-to-class="SharpViewPanel--expanded" leave-class="SharpViewPanel--expanded" leave-active-class="SharpViewPanel--collapsing"
	    leave-to-class="SharpViewPanel--collapsed">
		<iframe src="data:text/html;charset=utf-8$,SOME%20HTML%20MARKUP" height="100%" width="100%" frameborder="0" style="height: 100%; width: 100%;"></iframe>
	</div>
</div>
`;

exports[`view-panel mount ViewPanel 1`] = `
<div>
	<div class="SharpViewPanel__glasspane" style="display: none;"></div>
	<div class="SharpViewPanel" style="display: none;" enter-class="SharpViewPanel--collapsed" enter-active-class="SharpViewPanel--expanding" enter-to-class="SharpViewPanel--expanded" leave-class="SharpViewPanel--expanded" leave-active-class="SharpViewPanel--collapsing"
	    leave-to-class="SharpViewPanel--collapsed">
		<!---->
	</div>
</div>
`;
